sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -109,24 +109,15 @@ public class StorageBlobClient: PipelineClient, PagedCollectionDelegate {
                 baseUrl: baseUrl,
                 transport: URLSessionTransport(),
                 policies: [
-                    // Python: QueueMessagePolicy(),
-                    HeadersPolicy(),
-                    // Python: config.proxy_policy,
-                    UserAgentPolicy(),
-                    // Python: StorageContentValidation(),
-                    // Python: StorageRequestHook(**kwargs),
+                    UserAgentPolicy(forLibraryWithBundleIdentifier: "com.azure.AzureStorageBlob"),
```

Comment:
Is there a way to programmatically obtain the bundle identifier or access it with a compiler directive? I ask due to the recent PR where the bundle identifier for the demo app was changed from `.demo` to `.examples` and it broke the demo app. Same concern as when we had the static embedded version.

Reply:
I would expect that the bundle ID should basically never change once we have our first release, so I don't see that as much of a problem. That said, we can get the main app bundle, or we can get a bundle by its bundle ID. There is no way to get the "current framework" bundle. The best we can do is call `Bundle(for: AnyClass)` and pass a class that was defined in the bundle. That incurs a reflection penalty though, whereas looking it up by bundle ID doesn't.

If we plumbed that through, I imagine it might look like:

```swift
// If you want to explicitly name the class
UserAgentPolicy(for: AppConfigurationClient.self)

// If you want to look it up from self
UserAgentPolicy(for: type(of: self))
```

I don't think there's a way to avoid providing that parameter at the call site. I don't know if I'm necessarily convinced this is better, what do you think?

Reply:
I think I'm convinced that the first form is an improvement over specifying the bundle. I added some fixups to make the change.

Reply:
Implemented in f71eea8, you can see how this looks in 0fc5d5f